sixtyfps_runtime/corelib/items/text.rs

```diff
@@ -597,7 +599,11 @@ impl TextInput {
         self.delete_selection();
         let mut text: String = self.text().into();
         let cursor_pos = self.selection_anchor_and_cursor().1;
-        text.insert_str(cursor_pos, text_to_insert);
+        if text_to_insert.contains('\n') && self.single_line() {
+            text.insert_str(cursor_pos, &text_to_insert.replace('\n', " "));
```

Comment:
I think that maybe this replacement should be done only at rendering time. If somebody sets a text with newline characters on a `TextInput` element that has `single-line: true` then it should also work, not only when editing text (where this code is called).

Reply:
yeah, it works already with Qt. But without the actual replacement, we would need to do that replacement in a few more place to get it work with femtovg because it renders new line with a box, and nothing after it.